src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java

```diff
@@ -80,96 +119,105 @@ public String getLabel() {
+        this.namespace = Util.fixEmpty(namespace);
     }
 
+    @CheckForNull
     public String getCloud() {
         return cloud;
     }
 
     @DataBoundSetter
-    public void setCloud(String cloud) {
-        this.cloud = cloud;
+    public void setCloud(@CheckForNull String cloud) {
+        this.cloud = Util.fixEmpty(cloud);
     }
 
+    @CheckForNull
     public String getInheritFrom() {
         return inheritFrom;
     }
 
     @DataBoundSetter
-    public void setInheritFrom(String inheritFrom) {
-        this.inheritFrom = inheritFrom;
+    public void setInheritFrom(@CheckForNull String inheritFrom) {
+        this.inheritFrom = Util.fixEmpty(inheritFrom);
     }
 
+    @CheckForNull
     public List<ContainerTemplate> getContainers() {
         return containers;
     }
 
     @DataBoundSetter
-    public void setContainers(List<ContainerTemplate> containers) {
-        this.containers = containers;
+    public void setContainers(@CheckForNull List<ContainerTemplate> containers) {
+        this.containers = Util.fixNull(containers);
     }
 
+    @CheckForNull
     public List<TemplateEnvVar> getEnvVars() {
         return envVars == null ? Collections.emptyList() : envVars;
     }
 
     @DataBoundSetter
-    public void setEnvVars(List<TemplateEnvVar> envVars) {
+    public void setEnvVars(@CheckForNull List<TemplateEnvVar> envVars) {
         if (envVars != null) {
             this.envVars.clear();
             this.envVars.addAll(envVars);
         }
     }
 
+    @CheckForNull
     public YamlMergeStrategy getYamlMergeStrategy() {
         return yamlMergeStrategy;
     }
 
     @DataBoundSetter
-    public void setYamlMergeStrategy(YamlMergeStrategy yamlMergeStrategy) {
+    public void setYamlMergeStrategy(@CheckForNull YamlMergeStrategy yamlMergeStrategy) {
         this.yamlMergeStrategy = yamlMergeStrategy;
     }
 
+    @CheckForNull
     public List<PodVolume> getVolumes() {
         return volumes;
     }
 
     @DataBoundSetter
-    public void setVolumes(List<PodVolume> volumes) {
-        this.volumes = volumes;
+    public void setVolumes(@CheckForNull List<PodVolume> volumes) {
+        this.volumes = Util.fixNull(volumes);
     }
 
+    @CheckForNull
     public WorkspaceVolume getWorkspaceVolume() {
-        return workspaceVolume;
+        return workspaceVolume == null ? DescriptorImpl.defaultWorkspaceVolume : this.workspaceVolume;
     }
 
     @DataBoundSetter
-    public void setWorkspaceVolume(WorkspaceVolume workspaceVolume) {
-        this.workspaceVolume = workspaceVolume;
+    public void setWorkspaceVolume(@CheckForNull WorkspaceVolume workspaceVolume) {
+        this.workspaceVolume = (workspaceVolume == null || workspaceVolume.equals(DescriptorImpl.defaultWorkspaceVolume)) ? null : workspaceVolume;
     }
 
     public int getInstanceCap() {
-        return instanceCap;
+        return instanceCap;// == null ? DescriptorImpl.defaultInstanceCap : instanceCap;
```

Comment:
If you are not going to make a change, better to revert.

Reply:
Did not mean to commit that commented code. I will remove it.